snagsby/sources.py

```diff
@@ -56,41 +63,49 @@ def options(self):
             for k, v in parse_qs(self.url.query).items()
         }
 
+    def get_raw_data(self):
+        raise NotImplementedError("Please implement get_raw_data")
+
+    def get_data(self):
+        return sanitize(self.get_raw_data())
+
+
+class AWSSource(SnagsbySource):
     @property
     def region_name(self):
         return self.options.get('region')
 
-    def get_data(self):
-        return sanitize(self.get_raw_data())
+    def get_boto3_session(self, opts=None):
```

Comment:
💯 based on my other experiences with boto, I'm really glad to see this already moving to use sessions rather than ad-hoc calls, since it'll make things like adding timeouts much easier later on.

Reply:
Yeah, centralizing the configuration of the boto3 session has always worked well for us.